db_stress_tool/db_stress_listener.cc

```diff
@@ -65,16 +65,20 @@ UniqueIdVerifier::UniqueIdVerifier(const std::string& db_name)
       // Newly created is ok.
       // But FileSystem doesn't tell us whether non-existence was the cause of
       // the failure. (Issue #9021)
-      Status s2 = fs.FileExists(path_, opts, /*dbg*/ nullptr);
-      if (!s2.IsNotFound()) {
-        fprintf(stderr, "Error opening unique id file: %s\n",
-                s.ToString().c_str());
-        assert(false);
+      Status s2 = fs->FileExists(path_, opts, /*dbg*/ nullptr);
+      if (s2.IsNotFound()) {
+        std::unique_ptr<FSWritableFile> wf;
+        s = fs->NewWritableFile(path_, FileOptions(), &wf, /*dbg*/ nullptr);
```

Comment:
This is unnecessary according to the API docs of ReopenWritableFile

Reply:
Now that you pointed it out, I dug in a bit more into it and turns out the StressTest() constructor is deleting the "dbstress" directory under "/tmp/rocksdbtest-0/" directory because --destroy_db_initially is by default on.
In UniqueIdVerifier() constructor, ReOpenWritableFile() tries to create the file 
"/tmp/rocksdbtest-0/dbstress/.unique_ids" but it fails because the directory dbstress is missing, so it fails to create the "unique_ids"file and "data_file_writer_" becomes an invalid pointer, causing seg faults in ~UniqueIdVerifier() and Verify() in release builds and asserting on assert(false) in debug builds.

So, below command seg faults on both debug and release builds
"rm -rf /tmp/rocksdbtest*; db_stress --ops_per_thread=1000 --reopen=5"
Below command passes on both debug and release builds
"rm -rf /tmp/rocksdbtest*; db_stress --ops_per_thread=1000 --reopen=5 --destroy_db_initially=0"

So I added change to create the db directory if it is missing, so that ReOpenWritableFile() can work as expected and there is no need of NewWritableFile().